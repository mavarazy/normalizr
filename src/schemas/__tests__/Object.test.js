// eslint-env jest
import { fromJS } from 'immutable';
import { denormalize, normalize, schema } from '../../';

describe('ObjectSchema normalization', () => {
  test('normalizes an object', () => {
    const userSchema = new schema.Entity('user');
    const object = new schema.Object({
      user: userSchema
    });
    expect(normalize({ user: { id: 1, type: 'user' } }, object)).toMatchSnapshot();
  });

  test(`normalizes plain objects as shorthand for ${schema.Object.name}`, () => {
    const userSchema = new schema.Entity('user');
    expect(normalize({ user: { id: 1, type: 'user' } }, { user: userSchema })).toMatchSnapshot();
  });

  test('filters out undefined and null values', () => {
    const userSchema = new schema.Entity('user');
    const users = { foo: userSchema, bar: userSchema, baz: userSchema };
    const normalizedValue = normalize({ foo: {}, bar: { id: '1', type: 'bar' } }, users);
    expect(normalizedValue).toMatchSnapshot();
  });
});

describe('ObjectSchema denormalization', () => {
  test('denormalizes an object', () => {
    const userSchema = new schema.Entity('user');
    const object = new schema.Object({
      user: userSchema
    });
    const entities = {
      user: {
        1: { id: 1, name: 'Nacho' }
      }
    };
    expect(denormalize({ user: 1 }, object, entities)).toMatchSnapshot();
    expect(denormalize({ user: 1 }, object, fromJS(entities))).toMatchSnapshot();
    expect(denormalize(fromJS({ user: 1 }), object, fromJS(entities))).toMatchSnapshot();
  });

  test('denormalizes plain object shorthand', () => {
    const userSchema = new schema.Entity('user');
    const entities = {
      user: {
        1: { id: 1, name: 'Jane' }
      }
    };
    expect(denormalize({ user: 1 }, { user: userSchema, tacos: {} }, entities)).toMatchSnapshot();
    expect(denormalize({ user: 1 }, { user: userSchema, tacos: {} }, fromJS(entities))).toMatchSnapshot();
    expect(denormalize(fromJS({ user: 1 }), { user: userSchema, tacos: {} }, fromJS(entities))).toMatchSnapshot();
  });
});
