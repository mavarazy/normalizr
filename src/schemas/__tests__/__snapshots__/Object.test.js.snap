// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectSchema denormalization denormalizes an object 1`] = `
Object {
  "user": Object {
    "id": 1,
    "name": "Nacho",
  },
}
`;

exports[`ObjectSchema denormalization denormalizes an object 2`] = `
Object {
  "user": Immutable.Map {
    "id": 1,
    "name": "Nacho",
  },
}
`;

exports[`ObjectSchema denormalization denormalizes an object 3`] = `
Immutable.Map {
  "user": Immutable.Map {
    "id": 1,
    "name": "Nacho",
  },
}
`;

exports[`ObjectSchema denormalization denormalizes plain object shorthand 1`] = `
Object {
  "user": Object {
    "id": 1,
    "name": "Jane",
  },
}
`;

exports[`ObjectSchema denormalization denormalizes plain object shorthand 2`] = `
Object {
  "user": Immutable.Map {
    "id": 1,
    "name": "Jane",
  },
}
`;

exports[`ObjectSchema denormalization denormalizes plain object shorthand 3`] = `
Immutable.Map {
  "user": Immutable.Map {
    "id": 1,
    "name": "Jane",
  },
}
`;

exports[`ObjectSchema normalization filters out undefined and null values 1`] = `
Object {
  "entities": Object {
    "user": Object {
      "1": Object {
        "id": "1",
        "type": "bar",
      },
    },
  },
  "result": Object {
    "bar": "1",
  },
}
`;

exports[`ObjectSchema normalization normalizes an object 1`] = `
Object {
  "entities": Object {
    "user": Object {
      "1": Object {
        "id": 1,
        "type": "user",
      },
    },
  },
  "result": Object {
    "user": 1,
  },
}
`;

exports[`ObjectSchema normalization normalizes plain objects as shorthand for ObjectSchema 1`] = `
Object {
  "entities": Object {
    "user": Object {
      "1": Object {
        "id": 1,
        "type": "user",
      },
    },
  },
  "result": Object {
    "user": 1,
  },
}
`;
